ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -290,6 +290,17 @@ void validate() {
         }
     }
 
+    public void validateResultSetProperties() {
+        String givenDataFormat = this.getDataFormat();
+        if (givenDataFormat == null) {
+            this.setDataFormat(IngestionProperties.DataFormat.csv);
+        } else {
+            Ensure.isTrue(givenDataFormat.equals(IngestionProperties.DataFormat.csv.name()),
```

Comment:
I must be missing something here - why does this seem to require that the user-provided DataFormat is CSV?

Reply:
It's explained in the next line - ResultSet translates into a csv stream, so for them specifically it has to be csv

Reply:
I see the comment, but is that accurate? I'm still feeling like I'm missing something obvious.

Reply:
OK I think I understand - `ResultSet`s have to be of format `CSV`. This isn't documented in the SDK, though. The further we get from "toshb'a"/the business logic being documentation the better.

Reply:
I added a line of doc, WDYT?

Reply:
Looks great!